upload-async/Sources/App/Controllers/UploadController.swift

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2021 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Hummingbird
+import HummingbirdFoundation
+
+struct UploadController {
+    func addRoutes(to group: HBRouterGroup) {
+        group.post("/", options: .streamBody, use: self.upload)
+    }
+    
+    /// Handles raw bytes by saving them to disk. Note that a good practice
+    /// would be to save a reference to the bytes in a database and return
+    /// that object’s ID. That way the system can locate the file later
+    /// - Parameter request: `HBRequest`. If the `File-Name` header is set, then that name will be used as the file name on disk, otherwise a UUID will be used.
+    /// - Returns: A descriptive `String`
+    private func upload(_ request: HBRequest) async throws -> String {
+        guard request.body.stream != nil else { throw HBHTTPError(.unauthorized) }
+        let fileName = fileName(for: request)
+        let fileURL = try destinationPath(for: fileName, allowsOverwrite: true)
+        request.logger.info(.init(stringLiteral: "Uploading: \(fileURL)"))
+        try await request.fileIO.writeFile(contents: request.body,
+                                           path: fileURL.path,
+                                           context: request.context,
+                                           logger: request.logger)
+        return "Uploaded as \(fileName)"
```

Comment:
Maybe return as JSON `{"filename": "\(filename)"}`. You'd want to do that with the `JSONEncoder` as well, so you get the correct content-type header, so set `application.encoder = JSONEncoder()` and then return 
```swift
struct UploadResponse: ResponseEncodable {
    let filename: String
}
return UploadResponse(filename: "\(fileName)")
```

Reply:
Added